From 588ba7e8567c83db4b910ba1f6902b320f5e2d1c Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@freescale.com>
Date: Tue, 21 Jan 2014 13:03:41 +0800
Subject: [PATCH 440/442] ENGR00295903 usb: update more unit tests for 3.10

- Add kinds of unit test for 3.10, such as hub test, usb classes test,
performance test, etc.
- Rename the test name, in that case, we can know what is the test
through file name.

Signed-off-by: Peter Chen <peter.chen@freescale.com>
---
 test/doc/ut_um.toc                                 | 32 ++++----
 .../doc/030-Connected-During-Boot-Test.txt         | 61 ++++++++++++++++
 .../doc/031-Reboot-Connected-Auto-Test.txt         | 57 +++++++++++++++
 .../doc/032-Manually-Plug-In-Out-Test.txt          | 62 ++++++++++++++++
 test/mxc_usb_test/doc/033-Wakeup-Test.txt          | 62 ++++++++++++++++
 .../doc/034-Remote-Wakeup-Regression-Test.txt      | 59 +++++++++++++++
 test/mxc_usb_test/doc/035-Loadable-Module-Test.txt | 46 ++++++++++++
 .../doc/036-Loadable-Module-Regression-Test.txt    | 54 ++++++++++++++
 .../doc/037-Auto-Suspend-Mouse-Test.txt            | 41 +++++++++++
 .../doc/038-Auto-Suspend-Regression-Test.txt       | 50 +++++++++++++
 test/mxc_usb_test/doc/039-HUB-Test.txt             | 68 +++++++++++++++++
 test/mxc_usb_test/doc/040-System-Pm-Test.txt       | 55 ++++++++++++++
 test/mxc_usb_test/doc/041-USB-Host-Class-Test.txt  | 69 ++++++++++++++++++
 .../doc/042-Peripheral-Mode-Class-Test.txt         | 74 +++++++++++++++++++
 test/mxc_usb_test/doc/043-Performance-Test.txt     | 67 +++++++++++++++++
 test/mxc_usb_test/doc/044-DR-Mode-Test.txt         | 85 ++++++++++++++++++++++
 .../mxc_usb_test/doc/045-Device-Host-Only-Test.txt | 85 ++++++++++++++++++++++
 test/mxc_usb_test/doc/FSL-UT-USB-030.txt           | 61 ----------------
 test/mxc_usb_test/doc/FSL-UT-USB-031.txt           | 57 ---------------
 test/mxc_usb_test/doc/FSL-UT-USB-032.txt           | 62 ----------------
 test/mxc_usb_test/doc/FSL-UT-USB-033.txt           | 62 ----------------
 test/mxc_usb_test/doc/FSL-UT-USB-034.txt           | 58 ---------------
 test/mxc_usb_test/doc/FSL-UT-USB-035.txt           | 46 ------------
 test/mxc_usb_test/doc/FSL-UT-USB-036.txt           | 54 --------------
 test/mxc_usb_test/doc/FSL-UT-USB-037.txt           | 41 -----------
 test/mxc_usb_test/doc/FSL-UT-USB-038.txt           | 50 -------------
 test/mxc_usb_test/doc/readme.txt                   |  4 +-
 27 files changed, 1013 insertions(+), 509 deletions(-)
 create mode 100644 test/mxc_usb_test/doc/030-Connected-During-Boot-Test.txt
 create mode 100644 test/mxc_usb_test/doc/031-Reboot-Connected-Auto-Test.txt
 create mode 100644 test/mxc_usb_test/doc/032-Manually-Plug-In-Out-Test.txt
 create mode 100644 test/mxc_usb_test/doc/033-Wakeup-Test.txt
 create mode 100644 test/mxc_usb_test/doc/034-Remote-Wakeup-Regression-Test.txt
 create mode 100644 test/mxc_usb_test/doc/035-Loadable-Module-Test.txt
 create mode 100644 test/mxc_usb_test/doc/036-Loadable-Module-Regression-Test.txt
 create mode 100644 test/mxc_usb_test/doc/037-Auto-Suspend-Mouse-Test.txt
 create mode 100644 test/mxc_usb_test/doc/038-Auto-Suspend-Regression-Test.txt
 create mode 100644 test/mxc_usb_test/doc/039-HUB-Test.txt
 create mode 100644 test/mxc_usb_test/doc/040-System-Pm-Test.txt
 create mode 100644 test/mxc_usb_test/doc/041-USB-Host-Class-Test.txt
 create mode 100644 test/mxc_usb_test/doc/042-Peripheral-Mode-Class-Test.txt
 create mode 100644 test/mxc_usb_test/doc/043-Performance-Test.txt
 create mode 100644 test/mxc_usb_test/doc/044-DR-Mode-Test.txt
 create mode 100644 test/mxc_usb_test/doc/045-Device-Host-Only-Test.txt
 delete mode 100644 test/mxc_usb_test/doc/FSL-UT-USB-030.txt
 delete mode 100644 test/mxc_usb_test/doc/FSL-UT-USB-031.txt
 delete mode 100644 test/mxc_usb_test/doc/FSL-UT-USB-032.txt
 delete mode 100644 test/mxc_usb_test/doc/FSL-UT-USB-033.txt
 delete mode 100644 test/mxc_usb_test/doc/FSL-UT-USB-034.txt
 delete mode 100644 test/mxc_usb_test/doc/FSL-UT-USB-035.txt
 delete mode 100644 test/mxc_usb_test/doc/FSL-UT-USB-036.txt
 delete mode 100644 test/mxc_usb_test/doc/FSL-UT-USB-037.txt
 delete mode 100644 test/mxc_usb_test/doc/FSL-UT-USB-038.txt

diff --git a/test/doc/ut_um.toc b/test/doc/ut_um.toc
index 5896c09..10fd69c 100644
--- a/test/doc/ut_um.toc
+++ b/test/doc/ut_um.toc
@@ -148,22 +148,22 @@ include::../mxc_ssi_test/doc/FSL-UT-SSI-005.txt[]
 
 USB
 ---
-include::../mxc_usb_test/doc/FSL-UT-USB-001.txt[]
-include::../mxc_usb_test/doc/FSL-UT-USB-002.txt[]
-include::../mxc_usb_test/doc/FSL-UT-USB-004.txt[]
-include::../mxc_usb_test/doc/FSL-UT-USB-005.txt[]
-include::../mxc_usb_test/doc/FSL-UT-USB-006.txt[]
-include::../mxc_usb_test/doc/FSL-UT-USB-007.txt[]
-include::../mxc_usb_test/doc/FSL-UT-USB-008.txt[]
-include::../mxc_usb_test/doc/FSL-UT-USB-009.txt[]
-include::../mxc_usb_test/doc/FSL-UT-USB-011.txt[]
-include::../mxc_usb_test/doc/FSL-UT-USB-012.txt[]
-include::../mxc_usb_test/doc/FSL-UT-USB-013.txt[]
-include::../mxc_usb_test/doc/FSL-UT-USB-014.txt[]
-include::../mxc_usb_test/doc/FSL-UT-USB-015.txt[]
-include::../mxc_usb_test/doc/FSL-UT-USB-016.txt[]
-include::../mxc_usb_test/doc/FSL-UT-USB-017.txt[]
-include::../mxc_usb_test/doc/FSL-UT-USB-018.txt[]
+include::../mxc_usb_test/doc/030-Connected-During-Boot-Test.txt[]
+include::../mxc_usb_test/doc/031-Reboot-Connected-Auto-Test.txt[]
+include::../mxc_usb_test/doc/032-Manually-Plug-In-Out-Test.txt[]
+include::../mxc_usb_test/doc/033-Wakeup-Test.txt[]
+include::../mxc_usb_test/doc/034-Remote-Wakeup-Regression-Test.txt[]
+include::../mxc_usb_test/doc/035-Loadable-Module-Test.txt[]
+include::../mxc_usb_test/doc/036-Loadable-Module-Regression-Test.txt[]
+include::../mxc_usb_test/doc/037-Auto-Suspend-Mouse-Test.txt[]
+include::../mxc_usb_test/doc/038-Auto-Suspend-Regression-Test.txt[]
+include::../mxc_usb_test/doc/039-HUB-Test.txt[]
+include::../mxc_usb_test/doc/040-System-Pm-Test.txt[]
+include::../mxc_usb_test/doc/041-USB-Host-Class-Test.txt[]
+include::../mxc_usb_test/doc/042-Peripheral-Mode-Class-Test.txt[]
+include::../mxc_usb_test/doc/043-Performance-Test.txt[]
+include::../mxc_usb_test/doc/044-DR-Mode-Test.txt[]
+include::../mxc_usb_test/doc/045-Device-Host-Only-Test.txt[]
 
 V4L2
 ----
diff --git a/test/mxc_usb_test/doc/030-Connected-During-Boot-Test.txt b/test/mxc_usb_test/doc/030-Connected-During-Boot-Test.txt
new file mode 100644
index 0000000..97c28b4
--- /dev/null
+++ b/test/mxc_usb_test/doc/030-Connected-During-Boot-Test.txt
@@ -0,0 +1,61 @@
+FSL-UT-USB-030-Connected-Test
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+[cols=">s,6a",frame="topbot",options="header"]
+|====================================================================
+|Name | Description
+
+| Summary |
+USB Connected Test During Booting Up
+
+| Automated |
+No
+
+| Kernel Config Option |
+
+| Software Dependency |
+N/A
+
+| Non-default Hardware Configuration |
+N/A
+
+| Test Procedure |
+NOTICE: please check clkcount often using dump-clocks.sh
+
+1.1: With USB Disk connected at OTG Port
+/unit_tests/dump-clocks.sh | grep usb
+modprobe g_mass_storage file=/dev/mmcblk0p1 removable=1
+modprobe -r g_mass_storage
+/unit_tests/dump-clocks.sh | grep usb
+
+1.2: With USB Disk connected at Host 1 Port
+/unit_tests/dump-clocks.sh | grep usb
+
+1.3: With USB Cable connected with PC, load gadget module
+/unit_tests/dump-clocks.sh | grep usb
+modprobe g_mass_storage file=/dev/mmcblk0p1 removable=1
+/unit_tests/dump-clocks.sh | grep usb
+disconnet PC
+/unit_tests/dump-clocks.sh | grep usb
+
+1.4: With USB Cable connected with USB Charger, load gadget module
+modprobe g_mass_storage file=/dev/mmcblk0p1 removable=1
+/unit_tests/dump-clocks.sh | grep usb
+disconnect usb charger
+connect to pc
+
+1.5: With Micro B-To-A cable, but without anything connected
+/unit_tests/dump-clocks.sh | grep usb
+plug in usb disk
+plug out Micro B-To-A cable
+modprobe g_mass_storage file=/dev/mmcblk0p1 removable=1
+connect to pc
+disconnect with pc
+/unit_tests/dump-clocks.sh | grep usb
+
+| Expected Result |
+Without kernel dump, without clock mismatch
+
+|====================================================================
+
+<<<
diff --git a/test/mxc_usb_test/doc/031-Reboot-Connected-Auto-Test.txt b/test/mxc_usb_test/doc/031-Reboot-Connected-Auto-Test.txt
new file mode 100644
index 0000000..0b080ab
--- /dev/null
+++ b/test/mxc_usb_test/doc/031-Reboot-Connected-Auto-Test.txt
@@ -0,0 +1,57 @@
+FSL-UT-USB-031-Reboot-Test
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+[cols=">s,6a",frame="topbot",options="header"]
+|====================================================================
+|Name | Description
+
+| Summary |
+Reboot Test With U-disk Connected
+
+| Automated |
+YES
+
+| Kernel Config Option |
+
+| Software Dependency |
+N/A
+
+| Non-default Hardware Configuration |
+N/A
+
+| Test Procedure |
+1. Put Below Commands at init script (at the end of /etc/rc.d/rcS, etc)
+mount -t nfs -o nolock 10.192.244.61:/rootfs/wb /mnt/nfs
+cd /mnt/nfs/vte_mx63_d/
+source manual_test
+
+/root/test_reboot.sh
+
+NOTICE: 10.192.244.61:/rootfs/wb is the testing team VTE rootfs
+
+2. Put the below test_reboot.sh at /root/
+
+get_scsi.sh 4
+if [ $? -eq 0 ]
+then
+echo "aa" >> /root/reboot_count.txt
+echo "the time of reboot:"
+cat /root/reboot_count.txt | wc -l
+reboot
+else
+echo "the udisk can't be recognized"
+fi
+
+NOTICE: get_scsi.sh is the test application at VTE, it will judge
+if the u-disk is existed or not.
+
+3. Plug u-disk at USB port (Do another test with HUB connected)
+
+4. Power up the board
+
+| Expected Result |
+The continueous reboot test can run more than 1000 iterations.
+
+|====================================================================
+
+<<<
diff --git a/test/mxc_usb_test/doc/032-Manually-Plug-In-Out-Test.txt b/test/mxc_usb_test/doc/032-Manually-Plug-In-Out-Test.txt
new file mode 100644
index 0000000..94b8fe0
--- /dev/null
+++ b/test/mxc_usb_test/doc/032-Manually-Plug-In-Out-Test.txt
@@ -0,0 +1,62 @@
+FSL-UT-USB-032-Manually-Plug-In-Out-Test
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+[cols=">s,6a",frame="topbot",options="header"]
+|====================================================================
+|Name | Description
+
+| Summary |
+Manually Plug in/out Test
+
+| Automated |
+No
+
+| Kernel Config Option |
+
+| Software Dependency |
+N/A
+
+| Non-default Hardware Configuration |
+N/A
+
+| Test Procedure |
+2.1: Host/Device switch test and read/write test
+modprobe g_mass_storage file=/dev/mmcblk0p1 removable=1
+Plug in Micro B-To-A cable with u-disk connected
+/* Mount VTE ENV */
+mount -t nfs -o nolock 10.192.244.61:/rootfs/wb /mnt/nfs
+cd /mnt/nfs/vte_mx63_d
+source manual_test
+mkdir -p /mnt/sda1; mount /dev/sda1 /mnt/sda1
+bonnie\+\+ -d /mnt/sda1 -u 0:0 -s 64 -r 32
+dt of=/mnt/sda1/test_file bs=4k limit=128m passes=20
+Plug out usb disk
+plug in usb-disk
+plug out Micro B-To-A cable,
+plug in USB charger, plug out USB charger,
+plug in usb cable connected to PC.
+At Host PC do below four commands:
+mkdir -p /mnt/flash
+mount -t vfat /dev/sdx1 /mnt/flash
+sudo bonnie++ -d /mnt/flash -u 0:0 -s 10 -r 5
+sudo dt of=/mnt/flash/test_file bs=4k limit=63m passes=20
+
+Plug in Micro B-TO-A cable with u-disk connected
+Check clk refcount(/unit_tests/dump-clocks.sh | grep usb)
+
+2.2: Repeat Host/Device switch test
+s1: Booting With USB Disk connected at OTG Port
+s2: modprobe g_mass_storage file=/dev/mmcblk0p1 removable=1
+s3: plug out Micro B-TO-A cable
+s4: plug in usb cable connecting with PC
+s5: plug out usb cable
+s6: plug in Micro B-TO-A cable with u-disk connected
+s7: repeat s3-s6 10 times
+s8: check clk refcount (/unit_tests/dump-clocks.sh | grep usb)
+
+| Expected Result |
+No Error, No Clock Mismatch
+
+|====================================================================
+
+<<<
diff --git a/test/mxc_usb_test/doc/033-Wakeup-Test.txt b/test/mxc_usb_test/doc/033-Wakeup-Test.txt
new file mode 100644
index 0000000..3c70fb9
--- /dev/null
+++ b/test/mxc_usb_test/doc/033-Wakeup-Test.txt
@@ -0,0 +1,62 @@
+FSL-UT-USB-033-Wakeup-Test
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+[cols=">s,6a",frame="topbot",options="header"]
+|====================================================================
+|Name | Description
+
+| Summary |
+Kinds of USB Wakeup Tests
+
+| Automated |
+No
+
+| Kernel Config Option |
+
+| Software Dependency |
+N/A
+
+| Non-default Hardware Configuration |
+N/A
+
+| Test Procedure |
+1:
+/* Enable Serial Wakeup */
+echo enabled > /sys/class/tty/ttymxc0/power/wakeup
+echo mem > /sys/power/state
+plug in USB device, the system should not be waken up
+wakeup system using console
+The u-disk should be enumerated successfully
+2:
+plug out u-disk
+plug in mouse/keyboard at OTG port with micro b-to-a cable
+
+echo enabled > /sys/bus/platform/devices/20c9000.usbphy/power/wakeup
+echo enabled > /sys/bus/platform/devices/2184000.usb/power/wakeup
+echo enabled > /sys/bus/platform/devices/ci_hdrc.0/power/wakeup
+echo enabled > /sys/bus/usb/devices/usb2/power/wakeup
+echo enabled > /sys/bus/usb/devices/2-1/power/wakeup
+
+echo mem > /sys/power/state
+Testing usb wakeup function using plug in/out, micro b-to-a plug
+in/out, and remote wakeup.
+
+NOTICE: If micro b-to-a cable is plugged out, the usb2 will be
+removed, and the usb2 wakeup setting needs to be re-set before suspend.
+If usb device is plugged out, the 2-1 wakeup setting needs to
+be re-set before suspend.
+
+3:
+modprobe g_mass_storage file=/dev/mmcblk0p1 removable=1 (or ther gadgets)
+echo mem > /sys/power/state
+wakeup system with vbus
+echo disabled > /sys/bus/platform/devices/20c9000.usbphy/power/wakeup
+echo disabled > /sys/bus/platform/devices/ci_hdrc.0/power/wakeup
+usb cable plug in and micro-b-to-a plug in can't wake up
+
+| Expected Result |
+Can wakeup with wakeup setting, can't wakeup without wakeup setting
+
+|====================================================================
+
+<<<
diff --git a/test/mxc_usb_test/doc/034-Remote-Wakeup-Regression-Test.txt b/test/mxc_usb_test/doc/034-Remote-Wakeup-Regression-Test.txt
new file mode 100644
index 0000000..2e3dad5
--- /dev/null
+++ b/test/mxc_usb_test/doc/034-Remote-Wakeup-Regression-Test.txt
@@ -0,0 +1,59 @@
+FSL-UT-USB-034-Remote-Wakeup-Regression-Test
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+[cols=">s,6a",frame="topbot",options="header"]
+|====================================================================
+|Name | Description
+
+| Summary |
+Remote Wakeup Regression Test
+
+| Automated |
+Yes
+
+| Kernel Config Option |
+
+| Software Dependency |
+N/A
+
+| Non-default Hardware Configuration |
+N/A
+
+| Test Procedure |
+1. Prepare a USB host which featured with remote wakeup alternatively, You can
+use another i.mx board as USB device.
+
+2. Boot up both boards, and connect two boards with USB cable.
+
+3. At device side, do below commands:
+modprobe g_zero autoresume=3  max_autoresume=8 autoresume_interval_ms=1
+we can use the above three parameters to control device send resume signal time.
+Autoresume is the minimum number of seconds before sending resume signal,
+and the time before sending resume signal will automatically increase according
+to interval value, the endms is the max number of seconds before sending
+resume signal.
+
+4. At host side, first do below command:
+echo enabled > /sys/bus/platform/devices/20c9000.usbphy/power/wakeup
+echo enabled > /sys/bus/platform/devices/2184000.usb/power/wakeup
+echo enabled > /sys/bus/platform/devices/ci_hdrc.0/power/wakeup
+echo enabled > /sys/bus/usb/devices/usb2/power/wakeup
+echo enabled > /sys/bus/usb/devices/2-1/power/wakeup
+And you need build a test script to repeat standby the system like below:
+while [ 1 ] ;do echo mem  > /sys/power/state;sleep 5 ;done
+Then run the test script.
+
+5. Then USB device will enter the suspended state ,and autoresume
+milliseconds later,the device will send resume signal to wake up host.
+6. The expected behavior is: the host device be waked up, and there is not
+re-enumeration happens.
+7. The host will enter suspended state again,and test case will repeat step
+5 and step 6.
+8. Do the same test with HUB connected.
+
+| Expected Result |
+See setp 5-7.
+
+|====================================================================
+
+<<<
diff --git a/test/mxc_usb_test/doc/035-Loadable-Module-Test.txt b/test/mxc_usb_test/doc/035-Loadable-Module-Test.txt
new file mode 100644
index 0000000..51838fd
--- /dev/null
+++ b/test/mxc_usb_test/doc/035-Loadable-Module-Test.txt
@@ -0,0 +1,46 @@
+FSL-UT-USB-035-Loadable-Module-Test
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+[cols=">s,6a",frame="topbot",options="header"]
+|====================================================================
+|Name | Description
+
+| Summary |
+Loadable Module Test
+
+| Automated |
+No
+
+| Kernel Config Option |
+Device Drivers  --->
+	[*] USB support  --->
+			<M>   ChipIdea Highspeed Dual Role Controller
+			<M>   USB Gadget Support  --->
+			[*]   USB Physical Layer drivers  --->
+				<M>   Freescale MXS USB PHY support
+
+| Software Dependency |
+N/A
+
+| Non-default Hardware Configuration |
+N/A
+
+| Test Procedure |
+1. modprobe phy_mxs_usb; modprobe ci_hdrc_imx
+2. modprobe g_mass_storage file=/dev/mmcblk0p1 removable=1
+3. plug in usb cable connecting with PC
+4. plug out usb cable
+5. plug in Micro B-TO-A cable with u-disk connected
+6. plug out Micro B-TO-A cable
+7. Check clk refcount(/unit_tests/dump-clocks.sh | grep usb)
+8. modprobe -r phy_mxs_usb; modprobe -r g_mass_storage
+9. modprobe -r ci_hdrc_imx
+10. Do step 1-9 with u-disk connected during the boot.
+
+
+| Expected Result |
+No Error, No clock mismatch
+
+|====================================================================
+
+<<<
diff --git a/test/mxc_usb_test/doc/036-Loadable-Module-Regression-Test.txt b/test/mxc_usb_test/doc/036-Loadable-Module-Regression-Test.txt
new file mode 100644
index 0000000..8e2d1df
--- /dev/null
+++ b/test/mxc_usb_test/doc/036-Loadable-Module-Regression-Test.txt
@@ -0,0 +1,54 @@
+FSL-UT-USB-036-Loadable-Module-Regression-Test
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+[cols=">s,6a",frame="topbot",options="header"]
+|====================================================================
+|Name | Description
+
+| Summary |
+Loadable Module Regression Test
+
+| Automated |
+Yes
+
+| Kernel Config Option |
+Device Drivers  --->
+	[*] USB support  --->
+			<M>   ChipIdea Highspeed Dual Role Controller
+			<M>   USB Gadget Support  --->
+			<M>   Freescale MXS USB PHY support
+
+| Software Dependency |
+N/A
+
+| Non-default Hardware Configuration |
+N/A
+
+| Test Procedure |
+1. Run below test module load/unload script without any connection
+2. Run below test module load/unload script with u-disk connection
+
+----------------------------- test.sh----------------------------
+counter=1
+while [ 1 ]
+do
+modprobe phy_mxs_usb
+modprobe ci_hdrc_imx
+modprobe g_mass_storage file=/dev/mmcblk0p1 removable=1 (or other gadget)
+sleep 1
+sleep $( expr $RANDOM % 6)
+modprobe -r g_mass_storage
+modprobe -r ci_hdrc_imx
+modprobe -r phy_mxs_usb
+sleep 2
+counter=$(( $counter + 1 ))
+echo "the counter is $counter"
+echo "unload module!!!!!!!!!!"
+done
+
+| Expected Result |
+No Error, No clock mismatch
+
+|====================================================================
+
+<<<
diff --git a/test/mxc_usb_test/doc/037-Auto-Suspend-Mouse-Test.txt b/test/mxc_usb_test/doc/037-Auto-Suspend-Mouse-Test.txt
new file mode 100644
index 0000000..b618f76
--- /dev/null
+++ b/test/mxc_usb_test/doc/037-Auto-Suspend-Mouse-Test.txt
@@ -0,0 +1,41 @@
+FSL-UT-USB-037-Auto-Suspend-Test
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+[cols=">s,6a",frame="topbot",options="header"]
+|====================================================================
+|Name | Description
+
+| Summary |
+Auto suspend and remote wakeup test at system ative mode
+
+| Automated |
+No
+
+| Kernel Config Option |
+
+| Software Dependency |
+N/A
+
+| Non-default Hardware Configuration |
+N/A
+
+| Test Procedure |
+1. Put device as well as controller to low power mode
+echo auto > /sys/bus/usb/devices/1-1/power/control
+
+2. dump clk
+cat /sys/kernel/debug/clk/clk_summary | grep usb
+
+3. For hid device, press key to wakeup device
+cat /sys/kernel/debug/clk/clk_summary | grep usb
+
+NOTE: Since the auto-suspend timeout is small, please check
+clk count just finish pressing the key
+
+| Expected Result |
+The USB device as well as controller can enters low power mode
+and be waken up successfully
+
+|====================================================================
+
+<<<
diff --git a/test/mxc_usb_test/doc/038-Auto-Suspend-Regression-Test.txt b/test/mxc_usb_test/doc/038-Auto-Suspend-Regression-Test.txt
new file mode 100644
index 0000000..26aacbf
--- /dev/null
+++ b/test/mxc_usb_test/doc/038-Auto-Suspend-Regression-Test.txt
@@ -0,0 +1,50 @@
+FSL-UT-USB-038-Auto-Suspend-Test
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+[cols=">s,6a",frame="topbot",options="header"]
+|====================================================================
+|Name | Description
+
+| Summary |
+Using /sys to suspend and auto-resume USB device
+
+| Automated |
+Yes
+
+| Kernel Config Option |
+
+| Software Dependency |
+N/A
+
+| Non-default Hardware Configuration |
+N/A
+
+| Test Procedure |
+test_auto_suspend.sh 1-1
+---------------------------------------------------------------
+counter=1
+while [ 1 ]
+do
+
+echo auto > /sys/bus/usb/devices/$1/power/control
+
+sleep 1
+echo on > /sys/bus/usb/devices/$1/power/control
+
+counter=$(( $counter + 1 ))
+sleep 1
+echo "the counter is $counter"
+
+sleep 1
+
+done
+---------------------------------------------------------------
+
+| Expected Result |
+The USB device as well as controller can enters low power mode
+and be waken up successfully, besides, there is no reset/timeout
+/disconnect occur during the test
+
+|====================================================================
+
+<<<
diff --git a/test/mxc_usb_test/doc/039-HUB-Test.txt b/test/mxc_usb_test/doc/039-HUB-Test.txt
new file mode 100644
index 0000000..9282878
--- /dev/null
+++ b/test/mxc_usb_test/doc/039-HUB-Test.txt
@@ -0,0 +1,68 @@
+FSL-UT-USB-039-HUB-Test
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+[cols=">s,6a",frame="topbot",options="header"]
+|====================================================================
+|Name | Description
+
+| Summary |
+USB Test with HUB Connected
+
+| Automated |
+No
+
+| Kernel Config Option |
+
+| Software Dependency |
+N/A
+
+| Non-default Hardware Configuration |
+N/A
+
+| Test Procedure |
+NOTICE: please check clkcount often
+
+1.1: With USB Hub connected at OTG Port boots up
+1.1.1 The u-disk connected at HUB
+check clk refcount is 1
+plug out u-disk
+check clk refcount is 0
+plug out hub cable
+check clk refcount is 0
+plug out Micro-AB cable
+check clk refcount is 0
+
+
+1.1.2 Nothing is needed at HUB
+After booting up, and check if clock refcount is 0
+plug in usb mouse to see if works OK
+plug out usb mouse to see if clock refcount is 0
+plug out HUB to see if clock refcount is 0
+plug out Micro AB cable to see if clock refcount is 0
+
+1.2: Check u-disk connect/disconnect at hub
+Plug in HUB
+check clk refcount is 0
+plug in u-disk
+check clk refcount is 1
+plug out u-disk
+check clk refcount is 0
+re-do u-disk connect/disconnect 10 times, the u-disk needs to be
+recognized at every time.
+plug out Micro-AB cable with u-disk connected
+check clk refcount is 0
+
+1.3: With hub connected, do system suspend/resume
+u-disk is connected, do "echo mem > /sys/power/state"
+after resume, the u-disk should be existed.
+without u-disk connection, after system resume,
+the u-disk needs to be recognized after connection.
+
+1.4 Redo 1.1-1.3 with USB mouse
+
+| Expected Result |
+Without kernel dump, without clock mismatch
+
+|====================================================================
+
+<<<
diff --git a/test/mxc_usb_test/doc/040-System-Pm-Test.txt b/test/mxc_usb_test/doc/040-System-Pm-Test.txt
new file mode 100644
index 0000000..41e6bec
--- /dev/null
+++ b/test/mxc_usb_test/doc/040-System-Pm-Test.txt
@@ -0,0 +1,55 @@
+FSL-UT-USB-040-system-pm-Test
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+[cols=">s,6a",frame="topbot",options="header"]
+|====================================================================
+|Name | Description
+
+| Summary |
+USB Disk connected during system suspend/resume
+
+| Automated |
+Yes
+
+| Kernel Config Option |
+
+| Software Dependency |
+N/A
+
+| Non-default Hardware Configuration |
+N/A
+
+| Test Procedure |
+NOTICE: please check clkcount often using dump-clocks.sh
+
+Pre-test operation
+echo enabled > /sys/class/tty/ttymxc0/power/wakeup
+mkdir -p /mnt/nfs
+mount -t nfs -o nolock 10.192.244.6:/rootfs/wb /mnt/nfs
+cd /mnt/nfs/vte_mx63_hf/
+source ./manual_test
+
+1.1: With USB Disk connected and copy data during suspend/resume
+plug in usb disk
+bonnie\+\+ -d /media/sda1 -u 0:0 -s 256 -r 128 & sleep 3; i=0; while [ $i -lt 200 ]; do echo "$i iterations of suspend/resume has finished"; rtc_testapp_6 -T 50; let i=i+1; sleep 5; done
+i=0; while [ $i -lt 20000 ]; do echo "$i iterations of suspend/resume has finished"; rtc_testapp_6 -T 2; let i=i+1; sleep 2; done
+
+1.2: With USB mouse connected and do suspend/resume
+plug in usb mouse
+i=0; while [ $i -lt 20000 ]; do echo "$i iterations of suspend/resume has finished"; rtc_testapp_6 -T 2; let i=i+1; sleep 2; done
+
+1.3: With USB HUB connected and do suspend/resume
+plug in usb HUB
+i=0; while [ $i -lt 20000 ]; do echo "$i iterations of suspend/resume has finished"; rtc_testapp_6 -T 2; let i=i+1; sleep 2; done
+
+1.4: With USB HUB (u-disk connected) connected and do suspend/resume
+plug in usb HUB (with u-disk connected)
+i=0; while [ $i -lt 20000 ]; do echo "$i iterations of suspend/resume has finished"; rtc_testapp_6 -T 2; let i=i+1; sleep 2; done
+
+| Expected Result |
+Without error during suspend/resume, and the usb functions are still ok
+after finishing suspend/resume test.
+
+|====================================================================
+
+<<<
diff --git a/test/mxc_usb_test/doc/041-USB-Host-Class-Test.txt b/test/mxc_usb_test/doc/041-USB-Host-Class-Test.txt
new file mode 100644
index 0000000..434320a
--- /dev/null
+++ b/test/mxc_usb_test/doc/041-USB-Host-Class-Test.txt
@@ -0,0 +1,69 @@
+FSL-UT-USB-041-USB-Host-Class-Test
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+[cols=">s,6a",frame="topbot",options="header"]
+|====================================================================
+|Name | Description
+
+| Summary |
+Unit test for usb class test
+
+| Automated |
+No
+
+| Kernel Config Option |
+
+| Software Dependency |
+N/A
+
+| Non-default Hardware Configuration |
+N/A
+
+| Test Procedure |
+
+Mass storage and mouse has been tested at other tests.
+
+1. Test camera
+1.1: modprobe uvcvideo (if the module is not loaded automatically)
+
+1.2: Try to find video device
+Before the camera plug in:
+root@freescale ~$ ls /dev/video*
+/dev/video16  /dev/video17  /dev/video18  /dev/video19  /dev/video20
+
+After the camera plug in:
+root@freescale ~$ ls /dev/video*
+/dev/video    /dev/video16  /dev/video18  /dev/video20
+/dev/video0   /dev/video17  /dev/video19
+
+So the vedio device is /dev/video0
+
+1.3: Run unit test
+./unit_tests/mxc_v4l2_capture.out -uvc -f YUYV -d /dev/video0 -ow 320 -oh 240 -c 1000 test.yuv
+
+You can stop capture using CTRL+C, and use vooya to watch the capture.
+
+2. Test USB sound card
+2.1 modprobe snd_usb_audio (if it does not load automatically)
+
+2.2 root@imx6qdlsolo:~# aplay -l
+card 2: Pro [SB X-Fi Surround 5.1 Pro], device 0: USB Audio [USB Audio]
+Subdevices: 1/1
+Subdevice #0: subdevice #0
+card 2: Pro [SB X-Fi Surround 5.1 Pro], device 1: USB Audio [USB Audio #1]
+Subdevices: 1/1
+Subdevice #0: subdevice #0
+
+2.3 Play music
+cd $STREAM_PATH/alsa_stream_music
+aplay -Dhw:2 audio48k16S.wav (supported rate)
+aplay -D plug:hw:2,0 *.wav (non supported rate)
+
+2.4 Check the music is ok
+
+| Expected Result |
+No error during the operation
+
+|====================================================================
+
+<<<
diff --git a/test/mxc_usb_test/doc/042-Peripheral-Mode-Class-Test.txt b/test/mxc_usb_test/doc/042-Peripheral-Mode-Class-Test.txt
new file mode 100644
index 0000000..21e5530
--- /dev/null
+++ b/test/mxc_usb_test/doc/042-Peripheral-Mode-Class-Test.txt
@@ -0,0 +1,74 @@
+FSL-UT-USB-042-USB-Peripheral-Mode-Class-Test
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+[cols=">s,6a",frame="topbot",options="header"]
+|====================================================================
+|Name | Description
+
+| Summary |
+Test kinds of class at peripheral mode
+
+| Automated |
+No
+
+| Kernel Config Option |
+
+| Software Dependency |
+N/A
+
+| Non-default Hardware Configuration |
+N/A
+
+| Test Procedure |
+Since the mass storage has been tested during the role switch
+test, so it only tests g_ether and g_serial
+
+1: Test g_serial
+1.1 modprobe g_serial
+connect usb cable between board and pc, and you will see below message
+"g_serial gadget: high-speed config #2: CDC ACM config"
+
+1.2 Test data transfer from device to host
+run "cat /dev/ttyACM0" at host side
+run "echo "Happy New Year" > /dev/ttyGS0" at device side
+You should see "Happy New Year" at console of host side
+
+1.3 Test data transfer from host to device
+run "cat /dev/ttyACM0" at device side
+run "echo "Happy New Year" > /dev/ttyGS0" at host side
+You should see "Happy New Year" at console of device side
+
+2: Test g_ether
+2.1 modprobe g_ether qmult=10
+connect usb cable between board and pc, and you will see below message
+"g_ether gadget: high-speed config #1: CDC Ethernet (ECM)"
+
+2.2
+At device side:
+ifconfig usb0 10.0.0.2 up
+At host side:
+sudo ifconfig usb0 10.0.0.1 up
+
+At device side:
+ping 10.0.0.1, for ubuntu host, the ip for usb0 will lost
+after several seconds, re ifconfig usb0 at host side
+
+After ping can run more than 10 seconds, stop it
+
+2.3 Using iperf to do ether test
+
+At host side:
+iperf -s -i 5 &
+At device side:
+iperf -t 1000000 -c  10.0.0.1 -d -i 5 &
+
+2.4
+The connect should be there during the overnight test
+The speed is about 120Mbps.
+
+| Expected Result |
+There should be no error during the transfer
+
+|====================================================================
+
+<<<
diff --git a/test/mxc_usb_test/doc/043-Performance-Test.txt b/test/mxc_usb_test/doc/043-Performance-Test.txt
new file mode 100644
index 0000000..548aca3
--- /dev/null
+++ b/test/mxc_usb_test/doc/043-Performance-Test.txt
@@ -0,0 +1,67 @@
+FSL-UT-USB-043-USB-Performance-Test
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+[cols=">s,6a",frame="topbot",options="header"]
+|====================================================================
+|Name | Description
+
+| Summary |
+Performance Test
+
+| Automated |
+No
+
+| Kernel Config Option |
+
+| Software Dependency |
+N/A
+
+| Non-default Hardware Configuration |
+N/A
+
+| Test Procedure |
+For device mode, using g_ether to test
+For host mode, using mass storage to test (hard disk is needed)
+
+1: Test g_ether
+1.1 modprobe g_ether qmult=10
+connect usb cable between board and pc, and you will see below message
+"g_ether gadget: high-speed config #1: CDC Ethernet (ECM)"
+
+1.2
+At device side:
+ifconfig usb0 10.0.0.2 up
+At host side:
+sudo ifconfig usb0 10.0.0.1 up
+
+At device side:
+ping 10.0.0.1, for ubuntu host, the ip for usb0 will lost
+after several seconds, re ifconfig usb0 at host side
+
+After ping can run more than 10 seconds, stop it
+
+1.3 Using iperf to do ether test
+
+At host side:
+iperf -s -i 5 &
+At device side:
+iperf -t 1000000 -c  10.0.0.1 -d -i 5 &
+
+2 mass storage test
+2.1 block layer Read
+time -p dd if=/dev/sda2 of=/dev/null bs=4096 count=102400
+2.2 block layer write
+time -p dd if=/dev/zero of=/dev/sda2 bs=4096 count=102400
+2.3 filesystem layer
+iozone -a -i 1 -i 0 -U /media/sda2 -s 4096m -f  /media/sda2/iozone.test -b /mnt/tmp/iozone_usb_result.xls
+
+| Expected Result |
+g_ether: 15MB/s
+Block layer read: 21.1MB/s
+Block layer write: 17.0MB/s
+Filesystem layer (ext3): 19MB/s(R), 7.2MB/s(W)
+NOTES: for iozone, it should has file cache during the test.
+
+|====================================================================
+
+<<<
diff --git a/test/mxc_usb_test/doc/044-DR-Mode-Test.txt b/test/mxc_usb_test/doc/044-DR-Mode-Test.txt
new file mode 100644
index 0000000..1e02a8d
--- /dev/null
+++ b/test/mxc_usb_test/doc/044-DR-Mode-Test.txt
@@ -0,0 +1,85 @@
+FSL-UT-USB-044-USB-DR-MODE-Test
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+[cols=">s,6a",frame="topbot",options="header"]
+|====================================================================
+|Name | Description
+
+| Summary |
+Test peripheral-only and host-only mode through dt file
+
+| Automated |
+No
+
+| Kernel Config Option |
+
+| Software Dependency |
+N/A
+
+| Non-default Hardware Configuration |
+N/A
+
+| Test Procedure |
+Change dts file to support it (eg: arch/arm/boot/dts/imx6qdl-sabreauto.dtsi)
+
+1. Test dr_mode = "host"
+1.1 Rebuild dtb
+# modprobe -r ci_hdrc_imx; modprobe -r phy_mxs_usb; modprobe phy_mxs_usb && modprobe ci_hdrc_imx
+try if peripheral function is ok, the g_serial should not be inserted.
+
+1.2 Hotplug test in low power status without host wakeup support
+# echo mem > /sys/power/state
+Plugin u-disk into Host/OTG port
+Resume system by POWER switch and the u-disk should be found
+For ARD, use console wakeup
+# echo mem > /sys/power/state
+Plugout u-disk from Host/OTG port
+Resume system by POWER switch and the u-disk detach disconnet should be noticed
+Repeat this step 3 times
+1.3 USB mouse remote wakeup test
+Connect USB mouse to Host/OTG port
+# low_power_usb.sh; echo mem > /sys/power/state
+Click USB mouse to wakeup system
+Repeat this step to verify system can suspend/resume again
+1.4 Connect/disconnect wakeup
+# low_power_usb.sh; echo mem > /sys/power/state
+Plugin or plug out USB device from Host/OTG,
+System should be wakeup
+5. Special for OTG port
+# echo mem > /sys/power/state
+Then only plugin a microAB to female A cable(without device) to OTG port
+The system should NOT be wakeup
+Resume board by POWER switch and
+# echo mem > /sys/power/state
+Connect OTG port to PC
+The system should NOT be wakeup
+
+2 Test dr_mode = "peripheral"
+2.1 boot from rebuilt kernel and dtb and usb mouse at the board
+# modprobe -r ci_hdrc_imx; modprobe -r phy_mxs_usb; modprobe phy_mxs_usb && modprobe ci_hdrc_imx
+The mouse should not be recognized.
+
+2. Test g_mass_storage module
+In board
+# dd if=/dev/zero of=/var/storage.img bs=1M count=64
+# mkfs.vfat /var/storage.img
+# modprobe g_mass_storage file=/var/storage.img removable=1
+In Host PC:
+# mkdir -p /mnt/flash
+# umount /mnt/flash
+# mount -t vfat /dev/sdxx /mnt/flash (sdxx is the partition exported by g_file_storage in target board whose size is near 64M)
+# bonnie++ -d /mnt/flash -u 0:0 -s 32 -r 16
+# dt of=/mnt/flash/test_file bs=4k limit=63m passes=20
+Above commands should pass
+In board:
+# modprobe -r g_mass_storage
+# modprobe -r ci_hdrc_imx; modprobe -r phy_mxs_usb
+Remove the USB cable from OTG port
+3. Repeat steps 1 & 2
+
+| Expected Result |
+There should be no error during the test
+
+|====================================================================
+
+<<<
diff --git a/test/mxc_usb_test/doc/045-Device-Host-Only-Test.txt b/test/mxc_usb_test/doc/045-Device-Host-Only-Test.txt
new file mode 100644
index 0000000..63ec635
--- /dev/null
+++ b/test/mxc_usb_test/doc/045-Device-Host-Only-Test.txt
@@ -0,0 +1,85 @@
+FSL-UT-USB-045-USB-Device-Host-Only-Test
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+[cols=">s,6a",frame="topbot",options="header"]
+|====================================================================
+|Name | Description
+
+| Summary |
+Test peripheral-only and host-only mode by building out other driver
+
+| Automated |
+No
+
+| Kernel Config Option |
+
+| Software Dependency |
+N/A
+
+| Non-default Hardware Configuration |
+N/A
+
+| Test Procedure |
+Build out other driver through make menuconfig
+
+1. Test host-only driver
+1.1 Rebuild kernel
+# modprobe -r ci_hdrc_imx; modprobe -r phy_mxs_usb; modprobe phy_mxs_usb && modprobe ci_hdrc_imx
+try if peripheral function is ok, the g_serial should not be inserted.
+
+1.2 Hotplug test in low power status without host wakeup support
+# echo mem > /sys/power/state
+Plugin u-disk into Host/OTG port
+Resume system by POWER switch and the u-disk should be found
+For ARD, use console wakeup
+# echo mem > /sys/power/state
+Plugout u-disk from Host/OTG port
+Resume system by POWER switch and the u-disk detach disconnet should be noticed
+Repeat this step 3 times
+1.3 USB mouse remote wakeup test
+Connect USB mouse to Host/OTG port
+# low_power_usb.sh; echo mem > /sys/power/state
+Click USB mouse to wakeup system
+Repeat this step to verify system can suspend/resume again
+1.4 Connect/disconnect wakeup
+# low_power_usb.sh; echo mem > /sys/power/state
+Plugin or plug out USB device from Host/OTG,
+System should be wakeup
+1.5 Special for OTG port
+# echo mem > /sys/power/state
+Then only plugin a microAB to female A cable(without device) to OTG port
+The system should NOT be wakeup
+Resume board by POWER switch and
+# echo mem > /sys/power/state
+Connect OTG port to PC
+The system should NOT be wakeup
+
+2 Test Test peripheral-only driver
+2.1 boot from rebuilt kernel and connect usb mouse at the board
+# modprobe -r ci_hdrc_imx; modprobe -r phy_mxs_usb; modprobe phy_mxs_usb && modprobe ci_hdrc_imx
+The mouse should not be recognized.
+
+2.2 Test g_mass_storage module
+In board
+# dd if=/dev/zero of=/var/storage.img bs=1M count=64
+# mkfs.vfat /var/storage.img
+# modprobe g_mass_storage file=/var/storage.img removable=1
+In Host PC:
+# mkdir -p /mnt/flash
+# umount /mnt/flash
+# mount -t vfat /dev/sdxx /mnt/flash (sdxx is the partition exported by g_file_storage in target board whose size is near 64M)
+# bonnie++ -d /mnt/flash -u 0:0 -s 32 -r 16
+# dt of=/mnt/flash/test_file bs=4k limit=63m passes=20
+Above commands should pass
+In board:
+# modprobe -r g_mass_storage
+# modprobe -r ci_hdrc_imx; modprobe -r phy_mxs_usb
+Remove the USB cable from OTG port
+2.3 Repeat above steps
+
+| Expected Result |
+There should be no error during the test
+
+|====================================================================
+
+<<<
diff --git a/test/mxc_usb_test/doc/FSL-UT-USB-030.txt b/test/mxc_usb_test/doc/FSL-UT-USB-030.txt
deleted file mode 100644
index 97c28b4..0000000
--- a/test/mxc_usb_test/doc/FSL-UT-USB-030.txt
+++ /dev/null
@@ -1,61 +0,0 @@
-FSL-UT-USB-030-Connected-Test
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-[cols=">s,6a",frame="topbot",options="header"]
-|====================================================================
-|Name | Description
-
-| Summary |
-USB Connected Test During Booting Up
-
-| Automated |
-No
-
-| Kernel Config Option |
-
-| Software Dependency |
-N/A
-
-| Non-default Hardware Configuration |
-N/A
-
-| Test Procedure |
-NOTICE: please check clkcount often using dump-clocks.sh
-
-1.1: With USB Disk connected at OTG Port
-/unit_tests/dump-clocks.sh | grep usb
-modprobe g_mass_storage file=/dev/mmcblk0p1 removable=1
-modprobe -r g_mass_storage
-/unit_tests/dump-clocks.sh | grep usb
-
-1.2: With USB Disk connected at Host 1 Port
-/unit_tests/dump-clocks.sh | grep usb
-
-1.3: With USB Cable connected with PC, load gadget module
-/unit_tests/dump-clocks.sh | grep usb
-modprobe g_mass_storage file=/dev/mmcblk0p1 removable=1
-/unit_tests/dump-clocks.sh | grep usb
-disconnet PC
-/unit_tests/dump-clocks.sh | grep usb
-
-1.4: With USB Cable connected with USB Charger, load gadget module
-modprobe g_mass_storage file=/dev/mmcblk0p1 removable=1
-/unit_tests/dump-clocks.sh | grep usb
-disconnect usb charger
-connect to pc
-
-1.5: With Micro B-To-A cable, but without anything connected
-/unit_tests/dump-clocks.sh | grep usb
-plug in usb disk
-plug out Micro B-To-A cable
-modprobe g_mass_storage file=/dev/mmcblk0p1 removable=1
-connect to pc
-disconnect with pc
-/unit_tests/dump-clocks.sh | grep usb
-
-| Expected Result |
-Without kernel dump, without clock mismatch
-
-|====================================================================
-
-<<<
diff --git a/test/mxc_usb_test/doc/FSL-UT-USB-031.txt b/test/mxc_usb_test/doc/FSL-UT-USB-031.txt
deleted file mode 100644
index 1afc9c1..0000000
--- a/test/mxc_usb_test/doc/FSL-UT-USB-031.txt
+++ /dev/null
@@ -1,57 +0,0 @@
-FSL-UT-USB-031-Reboot-Test
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-[cols=">s,6a",frame="topbot",options="header"]
-|====================================================================
-|Name | Description
-
-| Summary |
-Reboot Test With U-disk Connected
-
-| Automated |
-YES
-
-| Kernel Config Option |
-
-| Software Dependency |
-N/A
-
-| Non-default Hardware Configuration |
-N/A
-
-| Test Procedure |
-1. Put Below Commands at init script (at the end of /etc/rc.d/rcS, etc)
-mount -t nfs -o nolock 10.192.244.61:/rootfs/wb /mnt/nfs
-cd /mnt/nfs/vte_mx63_d/
-source manual_test
-
-/root/test_reboot.sh
-
-NOTICE: 10.192.244.61:/rootfs/wb is the testing team VTE rootfs
-
-2. Put the below test_reboot.sh at /root/
-
-get_scsi.sh 4
-if [ $? -eq 0 ]
-then
-echo "aa" >> /root/reboot_count.txt
-echo "the time of reboot:"
-cat /root/reboot_count.txt | wc -l
-reboot
-else
-echo "the udisk can't be recognized"
-fi
-
-NOTICE: get_scsi.sh is the test application at VTE, it will judge
-if the u-disk is existed or not.
-
-3. Plug u-disk at USB port
-
-4. Power up the board
-
-| Expected Result |
-The continueous reboot test can run more than 1000 iterations.
-
-|====================================================================
-
-<<<
diff --git a/test/mxc_usb_test/doc/FSL-UT-USB-032.txt b/test/mxc_usb_test/doc/FSL-UT-USB-032.txt
deleted file mode 100644
index 94b8fe0..0000000
--- a/test/mxc_usb_test/doc/FSL-UT-USB-032.txt
+++ /dev/null
@@ -1,62 +0,0 @@
-FSL-UT-USB-032-Manually-Plug-In-Out-Test
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-[cols=">s,6a",frame="topbot",options="header"]
-|====================================================================
-|Name | Description
-
-| Summary |
-Manually Plug in/out Test
-
-| Automated |
-No
-
-| Kernel Config Option |
-
-| Software Dependency |
-N/A
-
-| Non-default Hardware Configuration |
-N/A
-
-| Test Procedure |
-2.1: Host/Device switch test and read/write test
-modprobe g_mass_storage file=/dev/mmcblk0p1 removable=1
-Plug in Micro B-To-A cable with u-disk connected
-/* Mount VTE ENV */
-mount -t nfs -o nolock 10.192.244.61:/rootfs/wb /mnt/nfs
-cd /mnt/nfs/vte_mx63_d
-source manual_test
-mkdir -p /mnt/sda1; mount /dev/sda1 /mnt/sda1
-bonnie\+\+ -d /mnt/sda1 -u 0:0 -s 64 -r 32
-dt of=/mnt/sda1/test_file bs=4k limit=128m passes=20
-Plug out usb disk
-plug in usb-disk
-plug out Micro B-To-A cable,
-plug in USB charger, plug out USB charger,
-plug in usb cable connected to PC.
-At Host PC do below four commands:
-mkdir -p /mnt/flash
-mount -t vfat /dev/sdx1 /mnt/flash
-sudo bonnie++ -d /mnt/flash -u 0:0 -s 10 -r 5
-sudo dt of=/mnt/flash/test_file bs=4k limit=63m passes=20
-
-Plug in Micro B-TO-A cable with u-disk connected
-Check clk refcount(/unit_tests/dump-clocks.sh | grep usb)
-
-2.2: Repeat Host/Device switch test
-s1: Booting With USB Disk connected at OTG Port
-s2: modprobe g_mass_storage file=/dev/mmcblk0p1 removable=1
-s3: plug out Micro B-TO-A cable
-s4: plug in usb cable connecting with PC
-s5: plug out usb cable
-s6: plug in Micro B-TO-A cable with u-disk connected
-s7: repeat s3-s6 10 times
-s8: check clk refcount (/unit_tests/dump-clocks.sh | grep usb)
-
-| Expected Result |
-No Error, No Clock Mismatch
-
-|====================================================================
-
-<<<
diff --git a/test/mxc_usb_test/doc/FSL-UT-USB-033.txt b/test/mxc_usb_test/doc/FSL-UT-USB-033.txt
deleted file mode 100644
index 3c70fb9..0000000
--- a/test/mxc_usb_test/doc/FSL-UT-USB-033.txt
+++ /dev/null
@@ -1,62 +0,0 @@
-FSL-UT-USB-033-Wakeup-Test
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-[cols=">s,6a",frame="topbot",options="header"]
-|====================================================================
-|Name | Description
-
-| Summary |
-Kinds of USB Wakeup Tests
-
-| Automated |
-No
-
-| Kernel Config Option |
-
-| Software Dependency |
-N/A
-
-| Non-default Hardware Configuration |
-N/A
-
-| Test Procedure |
-1:
-/* Enable Serial Wakeup */
-echo enabled > /sys/class/tty/ttymxc0/power/wakeup
-echo mem > /sys/power/state
-plug in USB device, the system should not be waken up
-wakeup system using console
-The u-disk should be enumerated successfully
-2:
-plug out u-disk
-plug in mouse/keyboard at OTG port with micro b-to-a cable
-
-echo enabled > /sys/bus/platform/devices/20c9000.usbphy/power/wakeup
-echo enabled > /sys/bus/platform/devices/2184000.usb/power/wakeup
-echo enabled > /sys/bus/platform/devices/ci_hdrc.0/power/wakeup
-echo enabled > /sys/bus/usb/devices/usb2/power/wakeup
-echo enabled > /sys/bus/usb/devices/2-1/power/wakeup
-
-echo mem > /sys/power/state
-Testing usb wakeup function using plug in/out, micro b-to-a plug
-in/out, and remote wakeup.
-
-NOTICE: If micro b-to-a cable is plugged out, the usb2 will be
-removed, and the usb2 wakeup setting needs to be re-set before suspend.
-If usb device is plugged out, the 2-1 wakeup setting needs to
-be re-set before suspend.
-
-3:
-modprobe g_mass_storage file=/dev/mmcblk0p1 removable=1 (or ther gadgets)
-echo mem > /sys/power/state
-wakeup system with vbus
-echo disabled > /sys/bus/platform/devices/20c9000.usbphy/power/wakeup
-echo disabled > /sys/bus/platform/devices/ci_hdrc.0/power/wakeup
-usb cable plug in and micro-b-to-a plug in can't wake up
-
-| Expected Result |
-Can wakeup with wakeup setting, can't wakeup without wakeup setting
-
-|====================================================================
-
-<<<
diff --git a/test/mxc_usb_test/doc/FSL-UT-USB-034.txt b/test/mxc_usb_test/doc/FSL-UT-USB-034.txt
deleted file mode 100644
index b47c885..0000000
--- a/test/mxc_usb_test/doc/FSL-UT-USB-034.txt
+++ /dev/null
@@ -1,58 +0,0 @@
-FSL-UT-USB-034-Remote-Wakeup-Regression-Test
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-[cols=">s,6a",frame="topbot",options="header"]
-|====================================================================
-|Name | Description
-
-| Summary |
-Remote Wakeup Regression Test
-
-| Automated |
-Yes
-
-| Kernel Config Option |
-
-| Software Dependency |
-N/A
-
-| Non-default Hardware Configuration |
-N/A
-
-| Test Procedure |
-1. Prepare a USB host which featured with remote wakeup alternatively, You can
-use another i.mx board as USB device.
-
-2. Boot up both boards, and connect two boards with USB cable.
-
-3. At device side, do below commands:
-modprobe g_zero autoresume=3  max_autoresume=8 autoresume_interval_ms=1
-we can use the above three parameters to control device send resume signal time.
-Autoresume is the minimum number of seconds before sending resume signal,
-and the time before sending resume signal will automatically increase according
-to interval value, the endms is the max number of seconds before sending
-resume signal.
-
-4. At host side, first do below command:
-echo enabled > /sys/bus/platform/devices/20c9000.usbphy/power/wakeup
-echo enabled > /sys/bus/platform/devices/2184000.usb/power/wakeup
-echo enabled > /sys/bus/platform/devices/ci_hdrc.0/power/wakeup
-echo enabled > /sys/bus/usb/devices/usb2/power/wakeup
-echo enabled > /sys/bus/usb/devices/2-1/power/wakeup
-And you need build a test script to repeat standby the system like below:
-while [ 1 ] ;do echo mem  > /sys/power/state;sleep 5 ;done
-Then run the test script.
-
-5. Then USB device will enter the suspended state ,and autoresume
-milliseconds later,the device will send resume signal to wake up host.
-6. The expected behavior is: the host device be waked up, and there is not
-re-enumeration happens.
-7. The host will enter suspended state again,and test case will repeat step
-5 and step 6.
-
-| Expected Result |
-See setp 5-7.
-
-|====================================================================
-
-<<<
diff --git a/test/mxc_usb_test/doc/FSL-UT-USB-035.txt b/test/mxc_usb_test/doc/FSL-UT-USB-035.txt
deleted file mode 100644
index 51838fd..0000000
--- a/test/mxc_usb_test/doc/FSL-UT-USB-035.txt
+++ /dev/null
@@ -1,46 +0,0 @@
-FSL-UT-USB-035-Loadable-Module-Test
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-[cols=">s,6a",frame="topbot",options="header"]
-|====================================================================
-|Name | Description
-
-| Summary |
-Loadable Module Test
-
-| Automated |
-No
-
-| Kernel Config Option |
-Device Drivers  --->
-	[*] USB support  --->
-			<M>   ChipIdea Highspeed Dual Role Controller
-			<M>   USB Gadget Support  --->
-			[*]   USB Physical Layer drivers  --->
-				<M>   Freescale MXS USB PHY support
-
-| Software Dependency |
-N/A
-
-| Non-default Hardware Configuration |
-N/A
-
-| Test Procedure |
-1. modprobe phy_mxs_usb; modprobe ci_hdrc_imx
-2. modprobe g_mass_storage file=/dev/mmcblk0p1 removable=1
-3. plug in usb cable connecting with PC
-4. plug out usb cable
-5. plug in Micro B-TO-A cable with u-disk connected
-6. plug out Micro B-TO-A cable
-7. Check clk refcount(/unit_tests/dump-clocks.sh | grep usb)
-8. modprobe -r phy_mxs_usb; modprobe -r g_mass_storage
-9. modprobe -r ci_hdrc_imx
-10. Do step 1-9 with u-disk connected during the boot.
-
-
-| Expected Result |
-No Error, No clock mismatch
-
-|====================================================================
-
-<<<
diff --git a/test/mxc_usb_test/doc/FSL-UT-USB-036.txt b/test/mxc_usb_test/doc/FSL-UT-USB-036.txt
deleted file mode 100644
index 8e2d1df..0000000
--- a/test/mxc_usb_test/doc/FSL-UT-USB-036.txt
+++ /dev/null
@@ -1,54 +0,0 @@
-FSL-UT-USB-036-Loadable-Module-Regression-Test
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-[cols=">s,6a",frame="topbot",options="header"]
-|====================================================================
-|Name | Description
-
-| Summary |
-Loadable Module Regression Test
-
-| Automated |
-Yes
-
-| Kernel Config Option |
-Device Drivers  --->
-	[*] USB support  --->
-			<M>   ChipIdea Highspeed Dual Role Controller
-			<M>   USB Gadget Support  --->
-			<M>   Freescale MXS USB PHY support
-
-| Software Dependency |
-N/A
-
-| Non-default Hardware Configuration |
-N/A
-
-| Test Procedure |
-1. Run below test module load/unload script without any connection
-2. Run below test module load/unload script with u-disk connection
-
------------------------------ test.sh----------------------------
-counter=1
-while [ 1 ]
-do
-modprobe phy_mxs_usb
-modprobe ci_hdrc_imx
-modprobe g_mass_storage file=/dev/mmcblk0p1 removable=1 (or other gadget)
-sleep 1
-sleep $( expr $RANDOM % 6)
-modprobe -r g_mass_storage
-modprobe -r ci_hdrc_imx
-modprobe -r phy_mxs_usb
-sleep 2
-counter=$(( $counter + 1 ))
-echo "the counter is $counter"
-echo "unload module!!!!!!!!!!"
-done
-
-| Expected Result |
-No Error, No clock mismatch
-
-|====================================================================
-
-<<<
diff --git a/test/mxc_usb_test/doc/FSL-UT-USB-037.txt b/test/mxc_usb_test/doc/FSL-UT-USB-037.txt
deleted file mode 100644
index b618f76..0000000
--- a/test/mxc_usb_test/doc/FSL-UT-USB-037.txt
+++ /dev/null
@@ -1,41 +0,0 @@
-FSL-UT-USB-037-Auto-Suspend-Test
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-[cols=">s,6a",frame="topbot",options="header"]
-|====================================================================
-|Name | Description
-
-| Summary |
-Auto suspend and remote wakeup test at system ative mode
-
-| Automated |
-No
-
-| Kernel Config Option |
-
-| Software Dependency |
-N/A
-
-| Non-default Hardware Configuration |
-N/A
-
-| Test Procedure |
-1. Put device as well as controller to low power mode
-echo auto > /sys/bus/usb/devices/1-1/power/control
-
-2. dump clk
-cat /sys/kernel/debug/clk/clk_summary | grep usb
-
-3. For hid device, press key to wakeup device
-cat /sys/kernel/debug/clk/clk_summary | grep usb
-
-NOTE: Since the auto-suspend timeout is small, please check
-clk count just finish pressing the key
-
-| Expected Result |
-The USB device as well as controller can enters low power mode
-and be waken up successfully
-
-|====================================================================
-
-<<<
diff --git a/test/mxc_usb_test/doc/FSL-UT-USB-038.txt b/test/mxc_usb_test/doc/FSL-UT-USB-038.txt
deleted file mode 100644
index 26aacbf..0000000
--- a/test/mxc_usb_test/doc/FSL-UT-USB-038.txt
+++ /dev/null
@@ -1,50 +0,0 @@
-FSL-UT-USB-038-Auto-Suspend-Test
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-[cols=">s,6a",frame="topbot",options="header"]
-|====================================================================
-|Name | Description
-
-| Summary |
-Using /sys to suspend and auto-resume USB device
-
-| Automated |
-Yes
-
-| Kernel Config Option |
-
-| Software Dependency |
-N/A
-
-| Non-default Hardware Configuration |
-N/A
-
-| Test Procedure |
-test_auto_suspend.sh 1-1
----------------------------------------------------------------
-counter=1
-while [ 1 ]
-do
-
-echo auto > /sys/bus/usb/devices/$1/power/control
-
-sleep 1
-echo on > /sys/bus/usb/devices/$1/power/control
-
-counter=$(( $counter + 1 ))
-sleep 1
-echo "the counter is $counter"
-
-sleep 1
-
-done
----------------------------------------------------------------
-
-| Expected Result |
-The USB device as well as controller can enters low power mode
-and be waken up successfully, besides, there is no reset/timeout
-/disconnect occur during the test
-
-|====================================================================
-
-<<<
diff --git a/test/mxc_usb_test/doc/readme.txt b/test/mxc_usb_test/doc/readme.txt
index 24340d3..0353ecd 100644
--- a/test/mxc_usb_test/doc/readme.txt
+++ b/test/mxc_usb_test/doc/readme.txt
@@ -1,5 +1,5 @@
 For old ARC driver (kernel version below 3.0.35), the test cases
 is from FSL-UT-USB-001.txt - FSL-UT-USB-029.txt (if existed).
 
-For new chipidea driver (kernel version above 3.5.7), the test cases
-is from FSL-UT-USB-030.txt.
+For new chipidea driver (kernel version above 3.10), the test cases
+is from 030-xxx.txt.
-- 
1.8.0

